common/src/main/java/com/wynntils/core/notifications/MessageContainer.java

```diff
@@ -9,21 +9,36 @@
 
 public class MessageContainer {
     protected TextRenderTask message;
+    private Integer iterations;
 
     public MessageContainer(String message) {
         this.message = new TextRenderTask(message, TextRenderSetting.DEFAULT);
+        this.iterations = 1;
     }
 
     public MessageContainer(TextRenderTask message) {
         this.message = message;
+        this.iterations = 1;
     }
 
     public TextRenderTask getRenderTask() {
         return message;
     }
 
-    public void editMessage(String newMessage) {
-        this.message.setText(newMessage);
+    public Integer getIterations() {
+        return iterations;
+    }
+
+    public void editMessage(String newMessage, Boolean incrementIterations) {
+        String sendableMessage;
+        if (incrementIterations) {
+            this.iterations++;
+            sendableMessage = newMessage + " §7[x" + this.iterations + "]";
```

Comment:
I dont like this. In this method, you edit the message. 

Create a new method, "incrementDuplicate", increment the counter there. Then make method to get the formatted string, appended with "[x2]".

Reply:
Atm, for example, `update` doesn't even work with this code

Reply:
To clarify, you should append the iteration "on the fly" in `getRenderTask`.